catkin_tools/jobs/utils.py

```diff
@@ -50,42 +50,67 @@ def get_env_loaders(package, context):
     return sources
 
 
+def merge_envs(job_env, overlay_envs):
+    '''
+    In the merged/linked case of single env, this function amounts to a straight
+    assignment, but a more complex merge is required with isolated result spaces,
+    since a package's build environment may require extending that of multiple
+    other result spaces.
+    '''
+    merge_path_values = {}
+
+    for overlay_env in overlay_envs:
+        for key, values_str in overlay_env.items():
+            if key.endswith('PATH'):
```

Comment:
This is sooooo fragile. Imagine something like `MY_CONFIG_PATH=file:///path/to/config.conf` and that this is set by someone else as `MY_CONFIG_PATH=file:///path/to/override_config.conf`. This function will not handle that correctly. Obviously this is a made up example which is unlikely to happen, but if this function is being applied in all configurations of the tools, then if it does happen it will be broken and the user cannot even work around it.

So, is this function only being used in `--env-cache` or is it always being called, regardless of the tool configuration?